Packs/AlienVault_OTX/Integrations/AlienVault_OTX_v2/AlienVault_OTX_v2.py

```diff
@@ -153,28 +172,109 @@ def create_pulse_by_ec(entry: dict) -> dict:
     return assign_params(**pulse_by_ec)
 
 
-def create_attack_pattern_relationships(client: Client, raw_response: dict, entity_a: str, entity_a_type: str):
+def extract_attack_ids(raw_response: dict):
+    """
+    extract the attack_ids field from the raw response if exists
+
+    Args:
+        raw_response: The response gotten from the server
+
+    :returns:
+        the attack_ids field if exists, otherwise, an empty string.
+    """
+    attack_id_field = dict_safe_get(raw_response, ['pulse_info', 'pulses'], [{}])
+    return dict_safe_get(([{}] if attack_id_field == [] else attack_id_field)[0], ['attack_ids'], [''])
+
+
+def relationships_manager(client: Client, entity_a: str, entity_a_type: str, indicator_type: str,
+                          indicator: str, field_for_passive_dns_rs: str, feed_indicator_type_for_passive_dns_rs: str):
+    """
+    manage the relationships creation
+
+    Args:
+        client: Client object with request
+        entity_a: str the first entity of the relationship
+        entity_a_type: str the type of the first entity
+        indicator_type: str the indicator type to get the related information by
+        entity_b_type: str the indicator to get the related information by
+
+    :returns:
+        a list of the relationships that were created
+    """
     relationships: list = []
 
+    if client.max_indicator_relationships != 0:
+        params = {'limit': str(client.max_indicator_relationships)}
```

Comment:
not sure I understand what will we gain from that?
both 'alienvault_get_related_hashes_by_indicator_command' and 'alienvault_get_related_urls_by_indicator_command' are designed to get a query params dict as an input, then why not build it where we're building it now and send it to them?

Reply:
Oh but this is not accurate, they were not designed to get the params arg, you added the params to these functions and that is exactly my point. Each of these functions receives as a parameter several command arguments. Each command should receive the relevant argument values in a "naive" way and create the relevant request on its own (not receiving parts of it from outside). The reason is that we might decide tomorrow we want to enhance the command (which can be run from xsoar cli) with some other argument that should go to the request params, then it will be messier to do that.